core/src/lib.rs

```diff
@@ -23,7 +28,7 @@ pub struct Client<F>
 where
     F: CallbacksPtr,
 {
-    raw: sys::tb_client_t,
+    raw: Arc<Mutex<sys::tb_client_t>>,
```

Comment:
There should be no need to use mutex there. In case `tb_client_t` became thread-unsafe, `Client` itself should become `!Sync` instead.